geopandas/geodataframe.py

```diff
@@ -1389,6 +1389,8 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
             and self._geometry_column_name in result.columns
             and any(isinstance(t, GeometryDtype) for t in result.dtypes)
         ):
+            # apply calls _constructor which resets geom col name to geometry
+            result._geometry_column_name = self._geometry_column_name
```

Comment:
I suppose this is fine (at least short term), but I am wondering to what extent this could also be solved by implementing a smarter `GeoDataFrame._constructor` (that function can in theory know the `self._geometry_column_name` and try to preserve this)? 
(but that's maybe something to look into *after* getting a basic version of `_constructor` implemented in https://github.com/geopandas/geopandas/pull/2060)

Reply:
You're totally right, this should be superfluous once `_constructor` is smarter - it will already need to know about `self._geometry_column_name` to get the current tests to pass, from my tests so far. Having said that, the other PR might not be resolved super quick, and perhaps this could be looked at after the fact? - I suspect there might be a couple of other cases like this too where improving `_constructor` will simplify matters